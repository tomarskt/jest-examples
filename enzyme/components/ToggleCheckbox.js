import React, { Component } from 'react';
import MyButton from './Mybutton'

class ToggleCheckbox extends Component {
  constructor(props) {
    super(props);
    this.state = { isChecked: false };
    this.onChange = this.onChange.bind(this);
  }

  onChange() {
    console.log(' ---------------------- this.onChange is invoked  ---------------------- ')
    this.setState({ isChecked: !this.state.isChecked });
  }

  render() {
    return (
      <div>
        <label>
          <input
            type="checkbox"
            checked={this.state.isChecked}
            onChange={this.onChange}
          />
          {this.state.isChecked ? 'On' : 'Off'}
        </label>
        <MyButton onClick={this.onChange} />
      </div>
    );
  }
}

export default ToggleCheckbox;