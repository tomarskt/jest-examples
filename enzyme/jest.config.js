module.exports = {
    "verbose": false,
    "bail": true,
    "testEnvironment": "jsdom",
    "setupFiles": ["<rootDir>/setup/shim.js", "<rootDir>/setup/index.js"],
    "snapshotSerializers": ["enzyme-to-json/serializer"]
}