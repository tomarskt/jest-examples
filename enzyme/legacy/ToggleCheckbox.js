// Copyright 2004-present Facebook. All Rights Reserved.

import React from 'react';
import MyButton from './Mybutton'
export default class ToggleCheckbox extends React.Component {

  constructor(props) {
    super(props);
    this.state = { isChecked: false };

    this.onChange = this.onChange.bind(this);
  }

  componentDidMount() {
    console.log('######## componentDidMount called  ##################')
  }
  
  onChange() {
    this.setState({ isChecked: !this.state.isChecked });
  }

  render() {
    return (
      <div>
        <label>
          <input
            type="checkbox"
            checked={this.state.isChecked}
            onChange={this.onChange}
            />
          {this.state.isChecked ? 'On':'Off'}
        </label>
        <MyButton onClick={this.onChange} />
      </div>
    );
  }
}
