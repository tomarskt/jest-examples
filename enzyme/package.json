{
  "scripts": {
    "test": "jest",
    "debug": "node --inspect-brk node_modules/.bin/jest  --runInBand  "
  },
  "dependencies": {
    "react": "^16.4.0",
    "react-dom": "^16.4.0",
    "prop-types": "^15.6.2"
  },
  "devDependencies": {
    "babel-plugin-transform-async-to-generator": "^6.24.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-polyfill": "^6.26.0",
    "babel-preset-es2015": "*",
    "babel-preset-react": "*",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.0.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^23.4.1",
    "react-router-dom": "^4.3.1",
    "react-test-renderer": "^16.4.0",
    "redux-form": "^7.3.0",
    "lerna-test-library": "^1.2.1"
  }
}
