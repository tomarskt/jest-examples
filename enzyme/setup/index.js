import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// to supress all warnings, uncomment below line;
// no need to install jsdom or setup jsom, as it's part of jest and 
// jest default to jsdom in testEnvironmenet settins. see package.json jest section
// process.env.NODE_ENV = 'production' 


Enzyme.configure({ adapter: new Adapter() });

