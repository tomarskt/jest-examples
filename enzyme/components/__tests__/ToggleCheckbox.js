import React from 'react';
import { shallow, mount } from 'enzyme';
import ToggleCheckbox from '../ToggleCheckbox';

describe('using enzyme', () => {

    it('shallow', () => {
        // because it's shallow rendered, its children won't be hydrated;
        // result snapshot has <ToggleCheckbox> <MyButton/></ToggleCheckbox> structure
        const shallowed = shallow(<ToggleCheckbox />);
        expect(shallowed).toMatchSnapshot();
    });
});