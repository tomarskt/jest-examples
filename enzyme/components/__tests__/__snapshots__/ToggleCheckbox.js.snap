// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using enzyme shallow 1`] = `
<div>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Off
  </label>
  <MyButton
    onClick={[Function]}
  />
</div>
`;
